### PR TITLE
Updated valve model

### PR DESCRIPTION
**Pull request by davideziviani**

### CODE REVIEW

**Comment by davideziviani** [comment]
@ibell Have you seen those updates?

File: PDSim/flow/flow_models.pxd [hunk 1]
```diff
@@ -36,7 +36,8 @@ cdef class IsentropicNozzleWrapper(FlowFunction):
 cpdef double FrictionCorrectedIsentropicNozzle(double A, State State_up, State State_down, double delta, int Type, double t = *, double ro = *)
 
 cdef class ValveModel(object):
-    cdef public double E,A_port,A_valve,d_valve,l_valve,a_valve,h_valve,rho_valve,d_port,m_eff,C_D,k_valve,x_stopper
+    # cdef public double E,A_port,A_valve,d_valve,l_valve,a_valve,h_valve,rho_valve,d_port,m_eff,C_D,k_valve,x_stopper
```
Comment (ibell) [inline PDSim/flow/flow_models.pxd]: Please remove this commented line, there is no need for it.

File: PDSim/flow/flow_models.pyx [hunk 2]
```diff
@@ -391,23 +391,26 @@ cdef class ValveModel(object):
     """
     
     #Give Cython type definitions for terms
+    # @cython.locals(d_valve=cython.double,d_port=cython.double,C_D=cython.double,
```
Comment (ibell) [inline PDSim/flow/flow_models.pyx]: Please remove these commented lines

File: PDSim/flow/flow_models.pyx [hunk 3]
```diff
@@ -418,7 +421,7 @@ cdef class ValveModel(object):
         self.x_stopper = x_stopper
         self.key_up = key_up
         self.key_down = key_down
-        self.x_tr = 0.25*(self.d_port**2/self.d_valve)
+        self.x_tr = x_tr #0.25*(self.d_port**2/self.d_valve)
```
Comment (ibell) [inline PDSim/flow/flow_models.pyx]: Why did you do this?

Comment (davideziviani) [inline PDSim/flow/flow_models.pyx]: In the original ValveClass, the reed valve modeled is not general enough especially if the reed valve has for example two ports in series. The calculation of I,k_valve and m_eff would then be different (I am facing this problem in one of the recip compressor I am working on). Also we should be able to specify k_valve starting from the natural frequency and bending mode. For these reasons, I made them as input to the model and adapted the recip example too. I am going to remove the commented lines.

**Comment by ibell** [comment]
Also, I'm not sure how easy this would be to do, but could you put together a Jupyter notebook that calls the valve model?

File: examples/recip_compressor.py [hunk 4]
```diff
@@ -79,33 +79,38 @@ def Compressor():
...
     x_stopper = 0.0018      #Stopper location [m]
-    
+
+    I=(d_valve*h_valve**3)/12  #Moment of Intertia for valve,[m^4]
+    k_valve=(6*E*I)/(a_valve**2*(3*l_valve-a_valve))    #Valve stiffness
+    m_eff=(1/3)*rho_valve*l_valve*d_valve*h_valve      #Effective mass of valve reeds
+    x_tr_suction = 0.25*(recip.d_suction**2/d_valve)
+    x_tr_discharge = 0.25*(recip.d_discharge**2/d_valve)
+
     #The suction valve parameters
```
Comment (davideziviani) [inline examples/recip_compressor.py]: @ibell : this is how it is done for a simple recip example. Alternatively I can create a function inside recip/core.py
It is quite straight forward. I just took them out from ValveClass. The example works nicely.

**Comment by davideziviani** [comment]
Certainly I can prepare a notebook on the ValveClass. It will be helpful also for our paper.

**Comment by ibell** [comment]
Sorry I haven't been following very well things happening in PDSim.  I'll
review your code tonight.

**Comment by ibell** [comment]
Can you please make the necessary changes so I can merge this PR?

**Comment by davideziviani** [comment]
I cleaned the commented lines. It should be good now.

**Pull request merged**